pylmod/tests/test_gradebook.py

```diff
@@ -805,13 +807,195 @@ def test_spreadsheet2gradebook_multi(self):
+        # Make sure that an error is raised if self.ASSIGNMENT_BODY
+        # is updated without updating this test too
+        for assignment in self.ASSIGNMENT_BODY['data']:
+            assert assignment['assignmentId'] != new_assignment_id
+
+        def handle_create(request, uri, headers):
+            """Assert that the create POST looks the way we expect it to"""
+            expected_post = {
+                "gradingSchemeType": "NUMERIC",
+                "name": "New homework",
+                "weight": 1.0,
+                "dueDateString": "12-15-2013",
+                "shortName": "Newrk",
+                "graderVisible": False,
+                "maxPointsTotal": expected_max_points,
+                "gradebookId": 1234
+            }
+            assert expected_post == json.loads(request.body)
+            body = json.dumps({'data': {'assignmentId': new_assignment_id}})
+            return 200, headers, body
+
+        httpretty.register_uri(
+            httpretty.POST,
+            '{0}assignment'.format(self.GRADEBOOK_REGISTER_BASE),
+            body=handle_create
+        )
+        self._register_get_gradebook()
+        self._register_get_assignments()
+        self._register_get_students()
+        self._register_multi_grade({'message': 'success'})
+
+        gradebook = GradeBook(self.CERT, self.URLBASE, self.GBUUID)
+
+        spreadsheet = [
+            {
+                'External email': 'a@example.com',
+                'New homework': 2.2,
+                'max_pts': max_points,
+                'normalize': normalize
+            }
+        ]
+        gradebook._spreadsheet2gradebook_multi(
+            csv_reader=spreadsheet,
+            email_field='External email',
+            non_assignment_fields=[
+                'External email', 'max_pts', 'normalize'
+            ],
+            approve_grades=False,
+            use_max_points_column=use_max_points_column,
+            max_points_column='max_pts',
+            normalize_column='normalize'
+        )
+        # Verify that we got the grades we expect
+        last_request = httpretty.last_request()
+        assert json.loads(last_request.body) == [
+            {
+                u'assignmentId': new_assignment_id,
+                u'studentId': 1,
+                u'numericGradeValue': 2.2,
+                u'mode': 2,
+                u'isGradeApproved': False
+            }
+        ]
+
+    @httpretty.activate
+    def test_no_max_points_column(self):
+        """
+        If the input spreadsheet has no columns matching the max points or
+        normalize column names, use the default max points value.
+        """
+        new_assignment_id = 3
+        # Make sure that an error is raised if self.ASSIGNMENT_BODY
+        # is updated without updating this test too
+        for assignment in self.ASSIGNMENT_BODY['data']:
+            assert assignment['assignmentId'] != new_assignment_id
+
+        def handle_create(request, uri, headers):
```

Comment:
Not quite, the `maxPointsTotal` value is different. I'd like to keep these two functions within the tests because the alternative is making a function which inputs maxPointsTotal and returns this function, which seems messier for just the two tests.